### PR TITLE
Replace CanGc::note() calls with arguments in `components/script/dom`

### PR DESCRIPTION
**Pull request by taniishkaaa**
Replaces `CanGc::note()` calls with arguments passed by callers as defined in #33683 

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are a part of #33683 
- [X] These changes do not require tests because they do not modify functionality

### CODE REVIEW

**Review by jdm — APPROVED** [review]
Looking good! Want to make one last set of changes as well?

File: components/script/dom/element.rs [hunk 1]
```diff
@@ -2353,6 +2353,7 @@ impl ElementMethods for Element {
                     rect.origin.y.to_f64_px(),
                     rect.size.width.to_f64_px(),
                     rect.size.height.to_f64_px(),
+                    CanGc::note(),
```
Comment (jdm) [inline components/script/dom/element.rs:2356]: If you modify Bindings.conf like https://github.com/servo/servo/blob/028026bebe992431ede7c57b7e3065616d859026/components/script/dom/bindings/codegen/Bindings.conf#L43, we can add CanGc arguments to the generated trait methods that you touch in this PR as well.

Comment (taniishkaaa) [reply components/script/dom/element.rs:2356]: sounds great! let me get on that :)

File: components/script/dom/bindings/codegen/Bindings.conf [hunk 2]
```diff
@@ -40,7 +40,7 @@ DOMInterfaces = {
 
 'Range': {
     'weakReferenceable': True,
-    'canGc': ['CreateContextualFragment', 'ExtractContents', 'CloneContents', 'CloneRange', 'SurroundContents'],
+    'canGc': ['CreateContextualFragment', 'ExtractContents', 'CloneContents', 'CloneRange', 'SurroundContents', 'ParseFromString', 'GetBounds', 'GetClientRects'],
```
Comment (jdm) [inline components/script/dom/bindings/codegen/Bindings.conf:43]: Sorry, I wasn't clear. Each canGc section in this file applies to a single interface name. You will need to add new sections for each interface you want to change in this way, then run a build to verify that the arguments are now present, and then add the missing arguments once the build fails.

Comment (taniishkaaa) [reply components/script/dom/bindings/codegen/Bindings.conf:43]: okay, i think i understand! so the `'weakReferenceable': True,` attribute should be there for all the new sections, right?

Comment (jdm) [reply components/script/dom/bindings/codegen/Bindings.conf:43]: No. Any interface that is not already present can have a new section added with only the canGc property present.

Comment (taniishkaaa) [reply components/script/dom/bindings/codegen/Bindings.conf:43]: also, is this a valid section relating to the changes in `domrect.rs`?
`'DOMRect': {
    'weakReferenceable': True,
    'canGc': ['new'],
},`

Comment (jdm) [reply components/script/dom/bindings/codegen/Bindings.conf:43]: Ah, no, it's only for methods that appear in the DOMRectMethods trait in this case.

Comment (taniishkaaa) [reply components/script/dom/bindings/codegen/Bindings.conf]: Can you please check if the changes are correct this time?

File: components/script/dom/bindings/codegen/Bindings.conf [hunk 3]
```diff
@@ -51,6 +51,14 @@ DOMInterfaces = {
     'weakReferenceable': True,
 },
 
+'DOMParser': {
+    'canGc': ['ParseFromString'],
+},
+
+'Element': {
+    'canGc': ['GetClientRects', 'GetBoundingClientRect'],
+},
```
Comment (jdm) [inline components/script/dom/bindings/codegen/Bindings.conf:60]: Oh, I see what happened. There are already DOMParser and Element sections in this file, so you should modify those instead.

Comment (taniishkaaa) [reply components/script/dom/bindings/codegen/Bindings.conf]: Done!

**Review by jdm — APPROVED** [review]
Thanks!

**Comment by mrobinson** [comment]
Looks like there are some merge conflicts with this PR now.

**Comment by taniishkaaa** [comment]
> Looks like there are some merge conflicts with this PR now.

I've rebased & pushed again. The conflicts should be resolved now.

**Pull request merged**